### PR TITLE
Fixes so far

### PR DESCRIPTION
**Pull request by CyberShadow**
This is a batch of fixes for issues that I have encountered while moving the bot away from Heroku and onto a self-hosted system, and while fixing the daily cron job task (which has been dysfunctional for a while).

Please see the commit list and their messages for details.

### CODE REVIEW

File: source/dlangbot/github_api.d [hunk 1]
```diff
@@ -34,9 +37,50 @@ struct Result
...
+    });
+    scope(exit) res.dropBody;
+
+    if (res.statusCode == 304 /*Not Modified*/)
+    {
+        logInfo(" > Cache hit!");
+        return cacheEntry;
+    }
+    else
+    if (res.statusCode == 200 /*OK*/)
```
Comment (Geod24) [inline source/dlangbot/github_api.d:64]: What's with the formatting ?

File: source/dlangbot/github_api.d [hunk 2]
```diff
@@ -34,9 +37,50 @@ struct Result
...
+        result.serializeToJsonString.toFile(cacheFileName);
+        return result;
+    }
+    else
+    if (res.statusCode / 100 == 3 && "Location" in res.headers)
+    {
+        auto location = res.headers["Location"];
+        logInfo(" > Redirect: %s", location);
+        return ghGetRequest(applyRelativeURL(url, location));
+    }
```
Comment (Geod24) [inline source/dlangbot/github_api.d:81]: This is not related to caching is it ?

**Review by Geod24 — COMMENTED** [review]
Nice. Just two questions, and I think you might want to update eventcore and vibe-d along with vibe-core, as many leaks have been fixed.

File: source/dlangbot/github_api.d (hunk 1 above)
Comment (CyberShadow) [reply source/dlangbot/github_api.d:64]: Sorry, what specifically? The `else\nif`?

File: source/dlangbot/github_api.d (hunk 2 above)
Comment (CyberShadow) [reply source/dlangbot/github_api.d:81]: Ah, no. I've reused code from dtest.dlang.io. The GitHub API does return redirects in some situations (e.g. https://api.github.com/repos/D-Programming-Language/dmd ) , so I think we do need to handle them in case something gets renamed or moved around in the future.

File: source/dlangbot/github_api.d (hunk 1 above)
Comment (CyberShadow) [reply source/dlangbot/github_api.d:64]: I see this is covered by https://dlang.org/dstyle.html#phobos_whitespace, so I'll fix it.

File: source/dlangbot/github_api.d (hunk 2 above)
Comment (Geod24) [reply source/dlangbot/github_api.d]: Could you split it in its own commit then ?

File: .d-compiler [hunk 3]
```diff
@@ -1 +1 @@
-dmd-2.085.0
```
Comment (Geod24) [inline .d-compiler:1]: BTW this file was for Heroku AFAIR so we can probably rm it

File: source/dlangbot/github_api.d (hunk 2 above)
Comment (CyberShadow) [reply source/dlangbot/github_api.d]: Sure, done.

File: .d-compiler (hunk 3 above)
Comment (CyberShadow) [reply .d-compiler:1]: That's OK, I use it for the scripts on the server with the same purpose.
